rsocket/ResumeManager.h

```diff
@@ -2,7 +2,73 @@
 
 #pragma once
 
+#include <folly/Optional.h>
+
+#include "rsocket/framing/Frame.h"
+#include "rsocket/framing/FrameTransport.h"
+
+namespace folly {
+class IOBuf;
+}
+
 namespace rsocket {
 
-class ResumeManager {};
+// Applications desiring to have cold-resumption should implement a
+// ResumeManager interface.  By default, an in-memory implementation
+// of this interface (InMemResumeManager) will be used by RSocket.
+class ResumeManager {
+ public:
+  virtual ~ResumeManager(){};
+
+  // The following methods will be called for each frame which is being
+  // sent/received on the wire.  The application should implement a way to
+  // store the sent and received frames in persistent storage.
+  virtual void trackReceivedFrame(
+      const folly::IOBuf& serializedFrame,
+      const FrameType frameType,
+      const StreamId streamId) = 0;
+
+  virtual void trackSentFrame(
+      const folly::IOBuf& serializedFrame,
+      const FrameType frameType,
+      const folly::Optional<StreamId> streamIdPtr) = 0;
+
+  // We have received acknowledgement from the remote-side that it has frames
+  // upto "position".  We can discard all frames before that.  This information
+  // is periodically received from remote-side through KeepAlive frames.
+  virtual void resetUpToPosition(ResumePosition position) = 0;
+
+  // The application should check its persistent storage and respond whether it
+  // has frames starting from "position" in send buffer.
```

Comment:
Maybe mention that `position` needs to explicitly be an element in the send buffer or the end, i.e. if the send buffer is [2,4,6,8], position=5 isn't valid even though it's part of the send buffer's bytes.